applications/blackhole/src/bh_context.erl

```diff
@@ -194,10 +199,18 @@ set_bindings(Context, Bindings) ->
 add_binding(#bh_context{bindings=Bds}=Context, Binding) ->
     Context#bh_context{bindings=[Binding|Bds]}.
 
+-spec add_bindings(context(), ne_binaries()) -> context().
+add_bindings(#bh_context{bindings=Bds}=Context, Bindings) ->
+    Context#bh_context{bindings= Bds ++ Bindings}.
+
 -spec remove_binding(context(), ne_binary()) -> context().
 remove_binding(#bh_context{bindings=Bds}=Context, Binding) ->
     Context#bh_context{bindings=lists:delete(Binding, Bds)}.
 
+-spec remove_bindings(context(), ne_binaries()) -> context().
+remove_bindings(#bh_context{bindings=Bds}=Context, Bindings) ->
+    Context#bh_context{bindings= Bds -- Bindings}.
```

Comment:
How about using a set instead of a list? Because if a binding appears multiple times in there this will potentially remove only as many as there are in `Bindings`